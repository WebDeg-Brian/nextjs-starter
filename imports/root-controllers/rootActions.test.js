import configureStore from 'redux-mock-store';
import {
  updateViewportDimensions,
  getBrowserInfo,
  getEngineInfo,
  getMobileInfo,
  getOSInfo,
} from './rootActions';
import {
  UPDATE_VIEWPORT_DIMENSIONS,
  GET_BROWSER_INFO,
  GET_ENGINE_INFO,
  GET_MOBILE_INFO,
  GET_OS_INFO,
} from './rootTypes';

const store = configureStore()();

describe('Check if actions return correct payloads', () => {
  beforeEach(() => {
    store.clearActions();
  });

  test('updateViewportDimensions', () => {
    const newWidth = 1024;
    const newHeight = 512;

    store.dispatch(updateViewportDimensions(newWidth, newHeight));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('getBrowserInfo', () => {
    const browserName = 'Firefox';
    const browserVersion = '63.0';
    const fullBrowserVersion = '63.0.1';

    store.dispatch(getBrowserInfo(browserName, browserVersion, fullBrowserVersion));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('getEngineInfo', () => {
    const engineName = 'Gecko';
    const engineVersion = '1.8.1';

    store.dispatch(getEngineInfo(engineName, engineVersion));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('getMobileInfo', () => {
    const mobileVendor = 'Apple';
    const mobileModel = 'iPhone 6s Plus';

    store.dispatch(getMobileInfo(mobileVendor, mobileModel));

    expect(store.getActions()).toMatchSnapshot();
  });

  test('getOSInfo', () => {
    const OSName = 'Windows';
    const OSVersion = '8.1';

    store.dispatch(getOSInfo(OSName, OSVersion));

    expect(store.getActions()).toMatchSnapshot();
  });
});
