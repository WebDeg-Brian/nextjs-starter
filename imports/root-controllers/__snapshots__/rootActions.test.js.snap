// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if actions return correct payloads getBrowserInfo 1`] = `
Array [
  Object {
    "browserName": "Firefox",
    "browserVersion": "63.0",
    "fullBrowserVersion": "63.0.1",
    "type": "GET_BROWSER_INFO",
  },
]
`;

exports[`Check if actions return correct payloads getEngineInfo 1`] = `
Array [
  Object {
    "engineName": "Gecko",
    "engineVersion": "1.8.1",
    "type": "GET_ENGINE_INFO",
  },
]
`;

exports[`Check if actions return correct payloads getMobileInfo 1`] = `
Array [
  Object {
    "mobileModel": "iPhone 6s Plus",
    "mobileVendor": "Apple",
    "type": "GET_MOBILE_INFO",
  },
]
`;

exports[`Check if actions return correct payloads getOSInfo 1`] = `
Array [
  Object {
    "OSName": "Windows",
    "OSVersion": "8.1",
    "type": "GET_OS_INFO",
  },
]
`;

exports[`Check if actions return correct payloads updateViewportDimensions 1`] = `
Array [
  Object {
    "newHeight": 512,
    "newWidth": 1024,
    "type": "UPDATE_VIEWPORT_DIMENSIONS",
  },
]
`;
